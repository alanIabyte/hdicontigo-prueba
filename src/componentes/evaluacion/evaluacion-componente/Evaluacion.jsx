import React from "react";
import PropTypes from "prop-types";
import IconoEstrella from "@material-ui/icons/StarRounded";

import {
  EnvolvedorEvaluacion,
  TituloEvaluacion,
  ContenedorCalificacion,
  ContenedorIconoEstrella,
} from "./Evaluacion.styled";

const Evaluacion = (props) => {
  const { titulo, evaluacion, asignarEvaluacion } = props;
  const elementos = [1, 2, 3, 4, 5];

  return (
    <EnvolvedorEvaluacion>
      <TituloEvaluacion>{titulo}</TituloEvaluacion>
      <ContenedorCalificacion>
        {elementos.map((indice) => (
          <ContenedorIconoEstrella
            key={indice}
            marcado={indice <= evaluacion}
            onClick={() => {
              asignarEvaluacion(indice);
            }}
          >
            <IconoEstrella style={{ fontSize: 40 }} />
          </ContenedorIconoEstrella>
        ))}
      </ContenedorCalificacion>
    </EnvolvedorEvaluacion>
  );
};

Evaluacion.propTypes = {
  titulo: PropTypes.string,
  evaluacion: PropTypes.number,
  asignarEvaluacion: PropTypes.func,
};

Evaluacion.defaultProps = {
  titulo: "",
  evaluacion: 0,
  asignarEvaluacion() {},
};

export default Evaluacion;
