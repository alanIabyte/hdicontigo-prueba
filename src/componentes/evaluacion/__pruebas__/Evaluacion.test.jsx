import React from "react";
import renderer from "react-test-renderer";
import Evaluacion from "../evaluacion-componente/Evaluacion";

describe("#Evaluacion", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<Evaluacion />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
