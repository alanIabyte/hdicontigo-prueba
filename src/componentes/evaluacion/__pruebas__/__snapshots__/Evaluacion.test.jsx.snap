// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Evaluacion se despliega correctamente 1`] = `
<div
  className="sc-fqkvVR sc-ikkxIA wQLUp ghjEaT"
>
  <div
    className="sc-fqkvVR sc-dAbbOL wQLUp kbSGie"
  >
    
  </div>
  <div
    className="sc-feUZmu dHrNBq"
  >
    <div
      className="sc-fUnMCh dArqPm"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "fontSize": 40,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
        />
      </svg>
    </div>
    <div
      className="sc-fUnMCh dArqPm"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "fontSize": 40,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
        />
      </svg>
    </div>
    <div
      className="sc-fUnMCh dArqPm"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "fontSize": 40,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
        />
      </svg>
    </div>
    <div
      className="sc-fUnMCh dArqPm"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "fontSize": 40,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
        />
      </svg>
    </div>
    <div
      className="sc-fUnMCh dArqPm"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "fontSize": 40,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27l4.15 2.51c.76.46 1.69-.22 1.49-1.08l-1.1-4.72 3.67-3.18c.67-.58.31-1.68-.57-1.75l-4.83-.41-1.89-4.46c-.34-.81-1.5-.81-1.84 0L9.19 8.63l-4.83.41c-.88.07-1.24 1.17-.57 1.75l3.67 3.18-1.1 4.72c-.2.86.73 1.54 1.49 1.08l4.15-2.5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
